src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -575,12 +601,13 @@ ShardLength(uint64 shardId)
 
 
 /*
- * NodeHasActiveShardPlacements returns whether any active shards are placed on this node
+ * NodeHasShardPlacements returns whether any active shards are placed on the group
+ * this node is a part of.
  */
 bool
-NodeHasActiveShardPlacements(char *nodeName, int32 nodePort)
+NodeHasShardPlacements(char *nodeName, int32 nodePort, bool onlyConsiderActivePlacements)
 {
-	const int scanKeyCount = 3;
+	const int scanKeyCount = (onlyConsiderActivePlacements ? 2 : 1);
```

Comment:
Agree that looking up is annoying but the function is pretty small so that's not a big concern yet, and if I moved this down what how would I size `ScanKeyData scanKey[]`? I suppose I could use a `maxScanKeyCount` but the danger of that getting out of sync seems higher than the inconvenience of not immediately knowing what the value of `scanKeyCount` is.